pkg/controller/pcideviceclaim/pcideviceclaim_controller.go

```diff
@@ -155,21 +127,32 @@ func (h Handler) enablePassthrough(pd *v1beta1.PCIDevice) error {
 }
 
 func (h Handler) disablePassthrough(pd *v1beta1.PCIDevice) error {
-	err := unbindDeviceFromDriver(pd.Status.Address, vfioPCIDriver)
-	if err != nil {
-		return err
+	errDriver := unbindDeviceFromDriver(pd.Status.Address, vfioPCIDriver)
+	errKubeVirt := h.removeHostDeviceFromKubeVirt(pd)
+	if errDriver != nil && errKubeVirt == nil {
+		return errDriver
 	}
-	err = h.removeHostDeviceFromKubeVirt(pd)
-	if err != nil {
-		return err
+	if errDriver == nil && errKubeVirt != nil {
+		return errKubeVirt
+	}
+	if errDriver != nil && errKubeVirt != nil {
+		msg := fmt.Sprintf("failed unbinding driver and also kubevirt: (%s, %s)", errDriver, errKubeVirt)
+		return errors.New(msg)
 	}
 	return nil
 }
 
 // This function unbinds the device with PCI Address addr from the given driver
 // NOTE: this function assumes that addr is on THIS NODE, only call for PCI addrs on this node
 func unbindDeviceFromDriver(addr string, driver string) error {
-	path := fmt.Sprintf("/sys/bus/pci/drivers/%s/unbind", driver)
+	driverPath := fmt.Sprintf("/sys/bus/pci/drivers/%s", driver)
+	// Check if device at addr is already bound to driver
+	_, err := os.Stat(fmt.Sprintf("%s/%s", driverPath, addr))
+	if err != nil {
+		logrus.Errorf("Device at address %s is not bound to driver %s", addr, driver)
+		return nil
+	}
+	path := fmt.Sprintf("%s/unbind", driverPath)
```

Comment:
please remove #165 `file.Close()`, the `defer fileClose()` will run in any return path